docs/source/development/pyrealm_build_data.md

```diff
@@ -0,0 +1,129 @@
+---
+
+# The `pyrealm_build_data` package
+
+The `pyrealm` repository includes both the `pyrealm` package and the
+`pyrealm_build_data` package. The `pyrealm_build_data` package contains datasets that
+are used in the `pyrealm` build process. This includes:
+
+* Example datasets that are used in the package documentation, such as simple spatial
+  datasets for showing the use of the P Model.
+* "Golden" datasets including a set of input data and output predictions from
+  previous implementations of `pyrealm` functionality. These are used to benchmark
+  `pyrealm` functionality within the testing framework (see below).
+
+Note that `pyrealm_build_data` is a source distribution only (`sdist`) component of
+`pyrealm`, so is not included in binary distributions (`wheel`), such as those available
+from PyPi.
+
+The package is organised into submodules that reflect the data use or previous
+implementation.
+
+## The `bigleaf` submodule
+
+This submodule contains benchmark outputs from the `bigleaf` package in `R`, which has
+been used as the basis for core hygrometry functions. The `bigleaf_conversions.R` R
+script runs a set of test values through `bigleaf`. The first part of the file prints
+out some simple test values that have been used in package doctests and then the second
+part of the file generates more complex benchmarking inputs that are saved, along with
+`bigleaf` outputs as `bigleaf_test_values.json`.
+
+Running `bigleaf_conversions.R` requires an installation of R along with the `jsonlite`
+and `bigleaf` packages, and the script can then be run from within the submodule folder
+as:
+
+```sh
+Rscript bigleaf_conversions.R
+```
+
+## The `rpmodel` submodule
+
+This submodule contains benchmark outputs from the `rpmodel` package in `R`, which has
+been used as the basis for initial development of the standard P Model.
+
+### Test inputs
+
+The `generate_test_inputs.py` file defines a set of constants for running P Model
+calculations and then defines a set of scalar and array inputs for the forcing variables
+required to run the P Model. The array inputs are set of 100 values sampled randomly
+across the ranges of plausible forcing value inputs in order to benchmark the
+calculations of the P Model implementation. All of these values are stored in the
+`test_inputs.json` file.
+
+It requires `python` and the `numpy` package and can be run as:
+
+```sh
+python generate_test_inputs.py
+```
+
+### Simple `rpmodel` benchmarking
+
+The `test_outputs_rpmodel.R` contains R code to run the test input data set, and store
+the expected predictions from the `rpmodel` package as `test_outputs_rpmodel.json`. It
+requires an installation of `R` and the `rpmodel` package and can be run as:
+
+```sh
+Rscript test_outputs_rpmodel.R
+```
+
+### Global array test
+
+The remaining files in the submodule are intended to provide a global test dataset for
+benchmarking the use of `rpmodel` on a global time-series - so using XYT data. It is
```

Comment:
What is XYT data? Does that mean 2D data plus time?

Reply:
It does, but I'll clarify.